File: unit03-hw/letterGrade.rb
#! /usr/bin/ruby
# letterGrade.rb translates a 0-100 based numeric grade to letter grade
# Author: Robert Lowman
################################################################

# Input:  numeric grade based on 0-100 scale
# Precondition: The grade is bewteen 0-100
# Output: The corresponding letter grade for the given grade

def yearCode(grade)
  grade = grade / 10
  case grade
  when 10, 9
    puts 'A'
  when 8
    puts 'B'
  when 7
    puts 'C'
  when 6
    puts 'D'
  else
    puts 'F'
  end
end
    

if __FILE__ == $0
   print "Enter the numeric grade (0-100): "
   grade = gets
   print "Letter grade is: "
   puts yearCode(grade)
end

File: unit03-hw/letterGrade.cpp
/* letterGrade.cpp takes a numeric grade and returns the corresponding 
 * letter grade
 * Author: Robert Lowman
 ****************************************************************/

#include <iostream>                      // interactive I/O
#include <string>                        // string class
using namespace std;

char letterCode(int grade);

int main() {
  cout << "\nEnter the numeric grade (0-100): "; // prompt
  int grade;
  cin >> grade;                            // read year
  cout << letterCode(grade) << endl;         // display its code
}


/***************************************************************
 * letterCode() converts a numeric grade into a letter grade. *
 * Receive: grade, an int.
 * PRE: grade is between 0-100.
 * Return: the letter grade to the corresponding numeric grade.
 */
char letterCode(int grade) {
  char result = '\0';
  switch (grade / 10) {
  case 10: case 9:
    result = 'A';
    break;
  case 8:
    result = 'B';
    break;
  case 7:
    result = 'C';
    break;
  case 6:
    result = 'D';
    break;
  default:
    result = 'F';
    break;
  }
  return result;
}
